### ISSUE TITLE
Problems with GLM4V forward passes

### ISSUE THREAD

**Issue by yifei-he** [verl-project/verl#4095]
Hi VeRL team,

Thank you so much for the great work and the prompt inclusion of support for newly released models including the GLM4V family. I am running experiments on GLM-4.1V-Thinking and I find that your implementation under verl/models/transformers/glm4v.py might have some issues.

1. Inconsistencies for the requirement of `position_ids` dimension: In the function `process_position_ids`, it requires `position_ids` to be 3-dimensional with the first dimension having size 4. From my understanding, this corresponds to the full multimiodal `position_ids` where `position_ids[0]` corresponds to text, and `position_ids[1:]` corresponds to vision. However, in `_custom_flash_attention_forward`, you have `assert position_ids.ndim == 2` without intermediate changes of `position_ids`. I suppose this part corresponds to only taking the text `position_ids`, so in `glm4v_attn_forward` before the `_custom_flash_attention_forward` calling, we probably need `position_ids = position_ids[0].squeeze(1)` or similar treatment. 
2. The current implementation does not seem to support `use_fused_kernels`. When `use_fused_kernels=True`, the function calling incurs either `forward_with_torch_backend` or `forward_with_triton_backend`. Thus, the output after calling `self.critic_module` in `_forward_micro_batch` of `dp_critic.py` only consists of `log_probs` and `entropy`, which does not include the necessary `logits`. When `use_fused_kernels=False`, it calls `forward_with_normal_backend` which does not have this issue.

Thank you for considering those issues!

**Comment by Kirrito-k423** [comment]
本轮定时清理检查发现，这个 issue 创建已超过 6 个月，并且创建后没有任何评论或公开引用，因此按社区 issue 定时清理标准关闭。

证据：
- zero_comment_zero_reference_created_at: 2025-11-12T03:05:29Z
- comments_count: 0
- timeline_public_references_found: 0
- Issue: https://github.com/verl-project/verl/issues/4095
- Comments API: https://api.github.com/repos/verl-project/verl/issues/4095/comments
- Timeline API: https://api.github.com/repos/verl-project/verl/issues/4095/timeline

如果作者或社区成员仍能提供新的复现、证据或需求背景，欢迎继续评论请求重开，或创建带有最新信息的新 issue。

English:
This scheduled triage found that this issue was created more than 6 months ago and has had no comments or public references since creation, so it is being closed under the scheduled stale-issue cleanup policy.

Evidence:
- zero_comment_zero_reference_created_at: 2025-11-12T03:05:29Z
- comments_count: 0
- timeline_public_references_found: 0
- Issue: https://github.com/verl-project/verl/issues/4095
- Comments API: https://api.github.com/repos/verl-project/verl/issues/4095/comments
- Timeline API: https://api.github.com/repos/verl-project/verl/issues/4095/timeline

If the author or another community member can provide a fresh reproduction, evidence, or updated requirement context, please comment to request reopening or open a new issue with the latest details.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
